### PR TITLE
Add "rb_define_class_under"(C-API) specs for errors

### PR DESCRIPTION
**Pull request by kachick**

### CODE REVIEW

File: optional/capi/class_spec.rb [hunk 1]
```diff
@@ -207,6 +207,33 @@
...
+    it "raises a TypeError when given a non class object to superclass" do
+      lambda { @s.rb_define_class_under(CApiClassSpecs,
+                                        "ClassUnder5",
+                                        Module.new)
+      }.should raise_error(TypeError)
+    end
+
+    ruby_version_is "2.3" do
+      it "raises a TypeError when given a mismatched class to superclass" do
+        CApiClassSpecs::ClassUnder6 = Class.new CApiClassSpecs::Super
```
Comment (eregon) [inline optional/capi/class_spec.rb]: Could you add parenthesis around the arguments to the constructor for clarity?

Comment (kachick) [inline optional/capi/class_spec.rb]: Sure.

**Comment by eregon** [comment]
Merged, thanks that's some good finding!

**Comment by kachick** [comment]
Thanks to merge!

**Pull request merged**